packages/arb-avm-cpp/avm_values/include/avm_values/buffer.hpp

```diff
@@ -210,25 +210,31 @@ class RawBuffer {
 class Buffer {
    public:
     std::shared_ptr<RawBuffer> buf;
+    uint64_t maxAccess;
 
-    Buffer(const RawBuffer& buffer) {
+    Buffer(const RawBuffer& buffer, uint64_t mx) {
         buf = std::make_shared<RawBuffer>(buffer);
+        maxAccess = mx;
     }
 
-    Buffer() { buf = std::make_shared<RawBuffer>(); }
+    Buffer() {
+        buf = std::make_shared<RawBuffer>(); 
+        maxAccess = 0;
+    }
 
     Buffer(const std::vector<uint8_t>& data) : Buffer() {
         for (uint64_t i = 0; i < data.size(); i++) {
             buf = std::make_shared<RawBuffer>(buf->set(i, data[i]));
         }
+        maxAccess = data.size();
```

Comment:
There is an off-by-one issue here.  In most of the code, `maxAccess` is the maximum index that was modified. But here, `data.size()` returns the maximum index in `data`, plus 1.

Obviously, either approach is OK, but we should be consistent.